src/sitemapper.cr

```diff
@@ -19,36 +19,47 @@ module Sitemapper
     setting host : String, example: "https://mysite.com"
     setting sitemap_host : String? = nil
     setting max_urls : Int32 = 500
-    setting storage : Symbol = :local
+    setting storage : Sitemapper::Storage.class = Sitemapper::LocalStorage
     setting compress : Bool = true
-    setting aws_config : Hash(String, String)? = nil
+    setting aws_config : AwsStorageConfig? = nil
   end
 
   def self.config
     Sitemapper.settings
   end
 
-  def self.build
-    builder = Sitemapper::Builder.new(config.host, config.max_urls, config.use_index)
-    # The following two lines are duplicated from self.build with a block, since
-    # the "with builder yield" is not working when just delegating and passing
-    # the block
-    with builder yield
-    builder.generate
-  end
-
-  def self.build(host : String, max_urls : Int32, use_index : Bool)
+  # Build your sitemaps. The block arg is an instance of `Sitemapper::Builder`.
+  # Args default to the configuration, but can be overriden.
+  # ```
+  # Sitemapper.build(max_urls: 20) do |builder|
+  #   builder.add("/").add("/about")
+  # end
+  # ```
+  def self.build(
+    host : String = config.host,
+    max_urls : Int32 = config.max_urls,
+    use_index : Bool = config.use_index
+  ) : Array(Hash(String, String))
     builder = Sitemapper::Builder.new(host, max_urls, use_index)
-    with builder yield
```

Comment:
This doesn't always work. It's even been mentioned a few times of removing it from Crystal. I like it, but it leads to some confusion especially if you don't know about the `itself` method on objects.